File: Shared/src/main/java/com/lenta/shared/models/core/Batch.java
package com.lenta.shared.models.core;

import java.util.Date;

import lombok.Getter;

public class Batch implements IBatch {

    @Getter private String batchNumber;
    @Getter private Manufacturer manufacturer;
    @Getter private Date bottlingDate;
    @Getter private int count;

    public Batch(String batchNumber, Manufacturer manufacturer, Date bottlingDate, int countInBatch)
    {
        this.batchNumber = batchNumber;
        this.manufacturer = manufacturer;
        this.bottlingDate = bottlingDate;
        this.count = countInBatch;
    }

    public static Batch setBottlingDate(Batch batch, Date bottlingDate)
    {
        return new Batch(batch.batchNumber, batch.manufacturer, bottlingDate, batch.count);
    }

}

File: Shared/src/main/java/com/lenta/shared/models/core/Batch.kt
package com.lenta.shared.models.core

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
data class Batch(
        val batchNumber: String,
        val count: String? = null,
        val dateOfProduction: Date? = null,
        val dateOfExpiration: Date? = null,
        val manufacturer: Manufacturer? = null
) : Parcelable {

    companion object {
        fun setBottlingDate(batch: Batch, dateOfProduction: Date, dateOfExpiration: Date? = null): Batch {
            return Batch(batch.batchNumber, batch.count, dateOfProduction, dateOfExpiration, batch.manufacturer)
        }
    }
}